static/js/entry/root.js

```diff
@@ -20,23 +20,32 @@ const store = configureStore()
 
 const rootEl = document.getElementById("container")
 
-const history = createBrowserHistory()
-const renderApp = Component => {
-  ReactDOM.render(
-    <AppContainer>
-      <Component store={store} history={history}>
-        {routes}
-      </Component>
-    </AppContainer>,
-    rootEl
-  )
-}
+const authRequiredEndpoint = "/auth_required/"
+if (
+  !SETTINGS.session_url &&
+  window.location.pathname !== authRequiredEndpoint
+) {
+  // user does not have the jwt cookie, they must go through login workflow first
+  window.location = authRequiredEndpoint
```

Comment:
do you think there's an easy way to do this redirect with react-router instead? this might be a helpful pattern: https://reacttraining.com/react-router/web/example/auth-workflow